crates/hirola-core/src/mixins.rs

```diff
@@ -163,3 +213,256 @@ pub mod model {
+///     let state = Signal::new(vec!["black-box".to_string()]);
+///
+///     let change_class = state.callback(move |st, _e:Event| {
+///         match st.get().to_vec()[0].as_str() {
+///             "black-box" => st.set(vec!["yellow-box".to_string()]),
+///             "yellow-box" => st.set(vec!["black-box".to_string()]),
+///             _ => panic!()
+///         }
+///     });
+/// 
+///     html! {
+///         <div class="" mixin:class=&class(&state)>
+///             <style>r##"
+///             .black-box {
+///                 background-color: black;
+///                 color: white;
+///             }
+///             .yellow-box {
+///                 background-color: yellow;
+///             }
+///             "##</style>
+///             <button on:click=change_class>"Click to change colour"</button>
+///         </div>
+///     }
+/// }
+/// ```
+pub fn class<'a>(signal: &'a Signal<Vec<String>>) -> Box<dyn Fn(DomNode) -> () + 'a> {
+    let signal = signal.clone();
+
+    let cb = move |node: DomNode| {
+        let el =node.unchecked_into::<Element>();
+        let signal = signal.clone();
+        let current_classes: Signal<Vec<String>> = Signal::new(vec![]);
+        
+        create_effect(move || {
+            let mut old_classes = current_classes.get().to_vec();
+            let mut new_classes = signal.get().to_vec();
+            new_classes.dedup();
+            let mut new_classes_clone = new_classes.clone();
+
+
+            //filter the ones that already rendered
+            for old_class in &old_classes.clone() {
+                for new_class in &new_classes_clone.clone() {
+                    if old_class == new_class {
+                        let old_index = old_classes.iter().position(|c| c == old_class).unwrap();
+                        let new_index = new_classes_clone.iter().position(|c| c ==new_class).unwrap();
+                        old_classes.remove(old_index);
+                        new_classes_clone.remove(new_index);
+                    }
+                }
+            }
+
+            //remove excess
+            for old_class in &old_classes {
+                el.class_list().remove_1(old_class).unwrap();
+            }
+
+            //add new class
+            for new_class in &new_classes_clone {
+                el.class_list().add_1(new_class).unwrap();
+            }
+
+            current_classes.set(new_classes.to_vec());
+        })
+    };
+    Box::new(cb)
+}
+
+///Mixin for reactive `src` attribute
+/// 
+/// ## Example
+/// ```rust, no_run
+/// pub fn reactive_img(_app: &HirolaApp) -> Dom {
+///     let state = Signal::new(String::from("./cat.jpg"));
+///     
+///     html! {
+///         <div>
+///             <img mixin:src=&src(&state) />
+///             <button on:click=move |_| state.set("./dog.jpg".to_string())>"Change to dog"</button>
+///         </div>
+///     }
+/// }
+/// ```
+pub fn src<'a>(signal: &'a Signal<String>) -> Box<dyn Fn(DomNode) -> () + 'a> {
```

Comment:
This doesn't make much sense. I havent tried it but I think it should work without a mixin ie:
```rust 
/// pub fn reactive_img(_app: &HirolaApp) -> Dom {
///     let state = Signal::new(String::from("./cat.jpg"));
///     
///     html! {
///         <div>
///             <img src=&state.get() />
///             <button on:click=move |_| state.set("./dog.jpg".to_string())>"Change to dog"</button>
///         </div>
///     }
/// }